docs/docs/development/hardware-integration/shift-registers.md

```diff
@@ -0,0 +1,150 @@
+---
+title: Configuring Shift Registers
+sidebar_label: Shift Registers
+---
+
+Shift registers are the recommended method of adding additional GPIO pins to MCUs and boards, when a standard matrix results in an insufficient number of keys. They are recommended because they simultaneously have very low power consumption and are quite cheap. This page serves as a (brief) introduction to shift registers, how to use them in your design, and how to configure ZMK to use them correctly.
+
+:::note
+This page assumes that the part you are using has the code 74HC595. Other shift registers can work as well, but this is the most commonly used one.
+:::
+
+:::tip
+To understand how shift registers work, we recommend reading through ["How does the 74HC595 Shift Register work?"](https://lastminuteengineers.com/74hc595-shift-register-arduino-tutorial/#how-does-the-74hc595-shift-register-work).
+:::
+
+## Design Guidelines
+
+The shift register output pins should be connected to the output pins of a keyboard matrix, while the input pins should be connected directly to the MCU/board. This is to allow the input pins to trigger "interrupts" on the MCU/board, upon which it will begin scanning the keys. Using a shift register for input pins is also possible, but requires you to use a different part and will harm your battery life.
```

Comment:
```suggestion
The shift register output pins should be connected to the output pins of a keyboard matrix, while the matrix input pins should be connected directly to the MCU/board. This is to allow the input pins to trigger "interrupts" on the MCU/board, upon which it will begin scanning the keys. Using a shift register for input pins is also possible, but requires you to use a different part and will harm your battery life.
```
I think this "input pins" is confusable with inputs to the shift register instead.

Reply:
Also, we should remark that these are SIPO shift registers, as a general term; either in this paragraph or in the note box with the part number.

Reply:
Rephrased this section a bunch, I think it's much clearer now.